src/employees/data.ts

```diff
@@ -0,0 +1,465 @@
+                      phone: 9773413721
+                    },
+                    {
+                      id: 95,
+                      name: 'Odette Sears',
+                      firstName: 'Odette',
+                      lastName: 'Sears',
+                      title: 'Senior Software Developer',
+                      active: true,
+                      hireDate: '2019-01-01',
+                      phone: 2648186576
+                    }
+                  ]
+                }
+              ]
+            },
+            {
+              id: 45,
+              name: 'Zelda Medina',
+              firstName: 'Zelda',
+              lastName: 'Medina',
+              title: 'QA Architect',
+              active: true,
+              hireDate: '2019-01-17',
+              phone: 5633596023
+            }
+          ]
+        }
+      ]
+    },
+    {
+      id: 3,
+      name: 'Priscilla Frank',
+      firstName: 'Priscilla',
+      lastName: 'Frank',
+      title: 'Chief Product Officer',
+      active: true,
+      hireDate: '2019-01-01',
+      phone: 2172805300,
+      reports: [
+        {
+          id: 4,
+          name: 'Ursula Holmes',
+          firstName: 'Ursula',
+          lastName: 'Holmes',
+          title: 'EVP, Product Strategy',
+          active: true,
+          hireDate: '2019-01-17',
+          phone: 3709838796
+        },
+        {
+          id: 24,
+          name: 'Melvin Carrillo',
+          firstName: 'Melvin',
+          lastName: 'Carrillo',
+          title: 'Director, Developer Relations',
+          active: true,
+          hireDate: '2019-01-15',
+          phone: 3444969555,
+          reports: [
+            {
+              id: 29,
+              name: 'Martha Chavez',
+              firstName: 'Martha',
+              lastName: 'Chavez',
+              title: 'Developer Advocate',
+              active: true,
+              hireDate: '2019-01-17',
+              phone: 1407727509
+            },
+            {
+              id: 30,
+              name: 'Oren Fox',
+              firstName: 'Oren',
+              lastName: 'Fox',
+              title: 'Developer Advocate',
+              active: true,
+              hireDate: '2019-01-01',
+              phone: 7142842408
+            },
+            {
+              id: 41,
+              name: 'Amos Barr',
+              firstName: 'Amos',
+              lastName: 'Barr',
+              title: 'Developer Advocate',
+              active: true,
+              hireDate: '2019-01-17',
+              phone: 9965878405
+            }
+          ]
+        }
+      ]
+    }
+];
+function parseData(data: any[]): Employee[] {
+    return data.map(record => {
+        const children = record.reports;
+        if (children) {
+            record.reports = parseData(children);
+        }
+
+        return {
+            ...record,
+            hireDate: new Date(record.hireDate)
+        };
+    });
+}
+export const data = parseData(rawData);
```

Comment:
It's just one extra step to return a fresh copy on each call to ensure it can't be modified externally:

```suggestion
export const data = () => parseData(rawData);
```